config/branch.go

```diff
@@ -0,0 +1,71 @@
+package config
+
+import (
+	"errors"
+	"strings"
+
+	format "gopkg.in/src-d/go-git.v4/plumbing/format/config"
+)
+
+var (
+	errBranchEmptyName     = errors.New("branch config: empty name")
+	errBranchInvalidRemote = errors.New("branch config: invalid remote")
+)
+
+// Branch contains information on the
+// local branches and which remote to track
+type Branch struct {
+	// Name name of branch
+	Name string
+	// Remote name of remote to track
+	Remote string
+	// Merge is the local refspec for the branch
+	Merge string
+
+	raw *format.Subsection
+}
+
+// Validate validates fields of branch
+func (b *Branch) Validate() error {
+	if b.Name == "" {
+		return errBranchEmptyName
+	}
+
+	if b.Merge != "" && !strings.HasPrefix(b.Merge, "refs/heads/") {
```

Comment:
Check this using [`ReferenceName.IsBranch`](https://godoc.org/gopkg.in/src-d/go-git.v4/plumbing#ReferenceName.IsBranch)